### PR TITLE
Add gpuCI build script

### PR DESCRIPTION
**Pull request by charlesbluca**
This PR, along with some internal setup with webhooks/permissions will add gpuCI support to Distributed.

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `black distributed` / `flake8 distributed` / `isort distributed`

### CODE REVIEW

**Comment by quasiben** [comment]
rerun tests

**Comment by charlesbluca** [comment]
rerun tests

**Comment by quasiben** [comment]
rerun tests

**Comment by jrbourbeau** [comment]
rerun tests

**Comment by jrbourbeau** [comment]
✅

**Review by jrbourbeau — COMMENTED** [review]
Several tests are being skipped for the following reason

```
13:50:20 distributed/deploy/tests/test_adaptive_core.py:36
13:50:20 distributed/deploy/tests/test_adaptive_core.py:36
13:50:20 distributed/deploy/tests/test_adaptive_core.py:36
13:50:20 distributed/deploy/tests/test_adaptive_core.py:36
13:50:20   /workspace/distributed/deploy/tests/test_adaptive_core.py:36: PytestUnknownMarkWarning: Unknown pytest.mark.asyncio - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/mark.html
13:50:20     @pytest.mark.asyncio
```

This is because `distributed` uses `pytest-asyncio` in its test suite, but `dask` doesn't. Since we're using [`dask`s CI environment file](https://github.com/rapidsai/dask-build-environment/blob/1e4aa081e38cd46f9606bc0170da2b7ad79e2572/Dockerfile#L12) for both `dask` and `distributed` gpuCI runs, this means that we would need to make sure all testing dependencies for `distributed` are included in `dask`'s CI environment file. We might consider building separate docker images for `dask` and `distributed`. 

As a temporary workaround I'd suggest we install `pytest-asyncio` directly in the `build.sh` script here

Also cc @raydouglass for visibility

File: continuous_integration/gpuci/build.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+env
+
+gpuci_logger "Check GPU usage"
+nvidia-smi
+
+gpuci_logger "Activate conda env"
+. /opt/conda/etc/profile.d/conda.sh
+conda activate dask
+
+gpuci_logger "Install distributed"
```
Comment (jrbourbeau) [inline continuous_integration/gpuci/build.sh:40]: I think adding something like

```
# TODO: Build a separate docker image for distributed.
# Manually install additional test dependencies as a temporary workaround.
gpuci_logger "Install additional testing dependencies"
mamba install -y -n dask -c conda-forge  "pytest-asyncio<0.14.0"
```

above this line should do the trick

**Comment by jrbourbeau** [comment]
Copying over a similar comment from https://github.com/dask/dask/pull/7966:

In the `conda list` step, our environment has the latest released version of `dask` installed. In general, the dev version of `distributed` depends on the dev version of `dask`. I'm not sure what the best way is to ensure we always have the latest dev version of `dask` installed. Perhaps add a `pip install git+https://github.com/dask/dask` line to `build.sh`?

**Review by jrbourbeau — APPROVED** [review]
I'm going to merge this PR in order to get the gpuCI build passing in other PRs. I'll open up a separate issue to track additional changes we may want to make. Thanks @charlesbluca!

**Pull request merged**